### PR TITLE
Hashicorp Vault external config supplier, plus documentation for external config

### PR DESCRIPTION
**Pull request by rdowner**
Adds an implementation of ExternalConfigSupplier for Hashicorp [Vault](https://www.vaultproject.io).

Also provides documentation for the externalized configuration system.

### CODE REVIEW

**Comment by ahgittin** [comment]
awesome -- and great docs!

File: core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultUserPassExternalConfigSupplier.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package org.apache.brooklyn.core.config.external.vault;
+
+import com.google.api.client.util.Lists;
```
Comment (sjcorbett) [inline core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultUserPassExternalConfigSupplier.java]: This import should be `com.google.common.collect.Lists`.

File: core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultAppIdExternalConfigSupplier.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
+package org.apache.brooklyn.core.config.external.vault;
+
+import com.google.api.client.util.Lists;
```
Comment (sjcorbett) [inline core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultAppIdExternalConfigSupplier.java]: And this one.

File: core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultExternalConfigSupplier.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
+package org.apache.brooklyn.core.config.external.vault;
+
+import com.google.api.client.util.Lists;
```
Comment (sjcorbett) [inline core/src/main/java/org/apache/brooklyn/core/config/external/vault/VaultExternalConfigSupplier.java]: And this one!

**Comment by ahgittin** [comment]
@sjcorbett good spot, i'll push a fix for those

**Comment by ahgittin** [comment]
also adding apache headers!

**Comment by rdowner** [comment]
Argh, rookie mistake... how long have I been here now? Thanks for fixing it up Alex.

**Pull request merged**